miden/src/cli/run.rs

```diff
@@ -0,0 +1,61 @@
+use super::data::{InputFile, OutputFile, ScriptFile};
+use air::StarkField;
+use log::debug;
+use std::path::PathBuf;
+use std::time::Instant;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "Run", about = "Run a miden program")]
+pub struct RunCmd {
+    /// Path to .masm script file
+    #[structopt(short = "s", long = "script-file", parse(from_os_str))]
+    script_file: PathBuf,
+    /// Path to input file
+    #[structopt(short = "i", long = "input-file", parse(from_os_str))]
+    input_file: Option<PathBuf>,
+    /// Number of ouptuts
+    #[structopt(short = "n", long = "num-outputs", default_value = "16")]
+    num_outputs: usize,
+    /// Path to output file
+    #[structopt(short = "o", long = "output-file", parse(from_os_str))]
+    output_file: Option<PathBuf>,
+}
+
+impl RunCmd {
+    pub fn execute(&self) -> Result<(), String> {
+        debug!("============================================================");
+        debug!("Run program");
+        debug!("============================================================");
+
+        // load script from file and compile
+        let script = ScriptFile::read(&self.script_file)?;
+
+        // load input data from file
+        let input_data = InputFile::read(&self.input_file, &self.script_file)?;
+
+        debug!("Executing program...");
+        let now = Instant::now();
+
+        // generate execution trace
+        let trace = processor::execute(&script, &input_data.get_program_inputs())
+            .map_err(|err| format!("Failed to generate exection trace = {:?}", err))?;
+
+        debug!(
+            "Program execution completed in {} ms",
+            now.elapsed().as_millis()
+        );
+
+        // extract outputs from execution trace
+        let outputs = trace.last_stack_state()[..self.num_outputs]
+            .iter()
+            .rev() // TODO investigate inconsistency - no reversal performed in comparison repo (see conditional.rs with n_outputs = 2)
```

Comment:
Sure thing I will change this.  In that case we will have to reverse the stack outputs after we read them from file.  The reason for this is because the `verify` interface requires the outputs in reversed order.

Reply:
I think we may need to come back to it later - but for now, reversing is fine. Maybe leave a comment there explaining this inconsistency.